### ISSUE TITLE
Overfitting in Transfer Learning

### ISSUE THREAD

**Issue by FranzHahn** [Skythianos/No-Reference-Video-Quality-Assessment-Based-on-the-Temporal-Pooling-of-Deep-Features#2]
The way you are applying Transfer Learning is incorrect. You are heavily overfitting on the training data, because you have data spill between the training and validation split created in your `transferLearningInceptionResNetV2.m` and `transferLearningInceptionV3.m` functions.

The following code causes the problem:

```
    images = imageDatastore(strcat(path,filesep,'SortedFrames'), 'IncludeSubfolders', true, 'LabelSource', 'foldernames');
    
    [trainFrames, valFrames] = splitEachLabel(images, 2/3, 'randomized');
```

This actually makes an even 2/3 - 1/3 split between all images on disk and loads them into the imageDatastore object. This doesn't consider the different videos as separate entities and consequently you have frames from the same videos show up in the training as well as the validation sets. Additionally, by training on this data immediately, you are not training on single random frames of the KoNViD-1k videos separately but on all frames of the videos at the same time.

What you would want to do is create separate train and validation folders within which you group the frames from videos selected as train or validation videos. All frames from videos that are selected as a training video should only be considered for training, all frames from videos that are selected as a validation video should only be considered for validation.

The fact that you are overfitting is immediately shown by your epoch 1 performance. It's unrealistic to have soemthing like 95% accuracy after a single epoch. As I have told you before, fixing this will result in approx. 45-50% validation accuracy after training for a few epochs.

Additionally, it will probably improve your generalization performance if you select one random frame of each training video each time you train and validate on all frames of the validation videos. For this to work best you should select the same number of frames of each video rather than a 1/5 chance on each individual frame as you have implemented currently.
